base_layer/core/src/proof_of_work/lwma_diff.rs

```diff
@@ -32,7 +38,25 @@ pub struct LinearWeightedMovingAverage {
 }
 
 impl LinearWeightedMovingAverage {
+    /// Initialize a new `LinearWeightedMovingAverage`
+    ///
+    /// # Panics
+    /// - Panics if block_window is 0
+    /// - Panics if `target_time * LWMA_MAX_BLOCK_TIME_RATIO != max_block_time`
     pub fn new(block_window: usize, target_time: u64, max_block_time: u64) -> Self {
+        assert!(
+            block_window > 0,
+            "TargetDifficulty::new expected block_window to be greater than 0, but 0 was given"
+        );
+        assert_eq!(
+            target_time * LWMA_MAX_BLOCK_TIME_RATIO,
+            max_block_time,
+            "LinearWeightedMovingAverage::new(...) expected max_block_time to be {} times greater than target_time, \
+             off by {}s",
+            LWMA_MAX_BLOCK_TIME_RATIO,
+            max(target_time * LWMA_MAX_BLOCK_TIME_RATIO, max_block_time) -
+                min(target_time * LWMA_MAX_BLOCK_TIME_RATIO, max_block_time)
+        );
```

Comment:
Using assert! and assert_eq! in prodection code is not ideal. Rather return an error here than a panic. 
Assertions are fine for tests, but this is not for tests